paddle/fluid/operators/gelu_op.cc

```diff
@@ -73,8 +84,19 @@ class GeluGradOp : public framework::OperatorWithKernel {
  protected:
   framework::OpKernelType GetExpectedKernelType(
       const framework::ExecutionContext &ctx) const override {
+    framework::LibraryType library{framework::LibraryType::kPlain};
+    framework::DataLayout layout = framework::DataLayout::kAnyLayout;
+#ifdef PADDLE_WITH_MKLDNN
+    auto it = this->Attrs().find("use_mkldnn");
+    if (library == framework::LibraryType::kPlain &&
+        it != this->Attrs().end() && platform::CanMKLDNNBeUsed(ctx)) {
+      library = framework::LibraryType::kMKLDNN;
+      layout = framework::DataLayout::kMKLDNN;
+    }
+#endif
     return framework::OpKernelType(
-        OperatorWithKernel::IndicateVarDataType(ctx, "X"), ctx.GetPlace());
+        OperatorWithKernel::IndicateVarDataType(ctx, "X"), ctx.GetPlace(),
```

Comment:
@iclementine Just a quick question I came up with during developing this PR. Shouldn't it be `IndicateVarDataType(ctx, "Out")` since Grad operator depends on `Out` variable?

Reply:
In this case, GeluGradOpMaker does not has `Out` as input, only `X` and `DOut`. So we use `X` to determine the data type of `DX`.

https://github.com/PaddlePaddle/Paddle/blob/dc4bc2624a2161887d941e32e7fb5fe1ca499e4f/paddle/fluid/operators/gelu_op.cc#L145